### PR TITLE
Cleanup Dragonfly-specific librustc_back target options

### PR DESCRIPTION
**Pull request by mneumann**
- Remove support for i686 (DragonFly dropped support of that architecture).
- Remove not-required target options.

### CODE REVIEW

**Comment by nagisa** [comment]
> Remove support for i686 (DragonFly dropped support of that architecture).

Not sure about this change. Surely there’s older dragonflies out there that still run on i686 and could run rust produced executables, aren’t there?

File: src/librustc_back/target/dragonfly_base.rs [hunk 1]
```diff
@@ -16,20 +16,10 @@ pub fn opts() -> TargetOptions {
         linker: "cc".to_string(),
         dynamic_linking: true,
         executables: true,
-        linker_is_gnu: true,
```
Comment (alexcrichton) [inline src/librustc_back/target/dragonfly_base.rs]: Just confirming, but it looks like the only place this is used is passing the `-O1` flag to the linker, and the dragonfly linker doesn't support that?

File: src/librustc_back/target/dragonfly_base.rs [hunk 2]
```diff
@@ -16,20 +16,10 @@ pub fn opts() -> TargetOptions {
...
         has_rpath: true,
-        pre_link_args: vec!(
-            "-L/usr/local/lib".to_string(),
-            "-L/usr/lib/gcc47".to_string(),
-            // GNU-style linkers will use this to omit linking to libraries
-            // which don't actually fulfill any relocations, but only for
-            // libraries which follow this flag.  Thus, use it before
-            // specifying libraries to link to.
-            "-Wl,--as-needed".to_string(),
```
Comment (alexcrichton) [inline src/librustc_back/target/dragonfly_base.rs]: This is actually pretty useful for slimming down unneeded dynamic dependencies if it's an accepted flag, so just confirming, this isn't accepted on dragonfly?

File: src/librustc_back/target/dragonfly_base.rs [hunk 3]
```diff
@@ -16,20 +16,10 @@ pub fn opts() -> TargetOptions {
...
-        ),
-        position_independent_executables: true,
```
Comment (alexcrichton) [inline src/librustc_back/target/dragonfly_base.rs]: Does dragonfly not support position independent executables? Or is the option perhaps named something else? (or is it the default?)

File: src/librustc_back/target/dragonfly_base.rs (hunk 1 above)
Comment (mneumann) [inline src/librustc_back/target/dragonfly_base.rs]: Hm, I wonder why FreeBSD (freebsd_base.rs) removed ```linker_is_gnu``` then. ```ld``` in DragonFly is gnu. So this clearly has to stay.

File: src/librustc_back/target/dragonfly_base.rs (hunk 2 above)
Comment (mneumann) [inline src/librustc_back/target/dragonfly_base.rs]: Accepted.

File: src/librustc_back/target/dragonfly_base.rs (hunk 3 above)
Comment (mneumann) [inline src/librustc_back/target/dragonfly_base.rs]: Again wondering why FreeBSD has removed this flag. They are supported.

**Comment by mneumann** [comment]
@nagisa: I doubt there are any i686 DragonFly's out there as it's depreciation has been announced quite some time ago, and even long before that, no one really used it. For sure, it doesn't hurt keeping support for it in Rust, but on the other hand, supporting an unmaintained (and as such potential insecure) platform might not be the right choice.

**Comment by mneumann** [comment]
Closing this pull request for now. I think the only thing that should be removed are the paths  "-L/usr/local/lib" and      "-L/usr/lib/gcc47" from ```pre_link_args```, as they will interfere with already installed Rust versions (can't compile rustc if rustc is installed in /usr/local).

**Comment by alexcrichton** [comment]
Ah ok, feel free to submit a PR regardless to remove those options, they do indeed seem a little sketchy!

Also if you want to add some of the flags and such to the FreeBSD options again feel free to do so as well!

**Comment by mneumann** [comment]
@alexcrichton I am not sure what the exact reasoning is behind the FreeBSD options. My pure guess is that they use the ```gold``` linker, which is not 100% ```ld``` compatible. I can't test atm.

**Pull request closed without merging**